packages/calcite-components/src/components/sheet/sheet.tsx

```diff
@@ -0,0 +1,365 @@
+import { LogicalFlowPosition, Scale } from "../interfaces";
+import { CSS_UTILITY } from "../../utils/resources";
+
+@Component({
+  tag: "calcite-sheet",
+  styleUrl: "sheet.scss",
+  shadow: true,
+})
+export class Sheet implements OpenCloseComponent, FocusTrapComponent, LoadableComponent {
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   *  Passes a function to run before the component closes.
+   *
+   * @returns {Promise<void>}
+   */
+  @Prop() beforeClose: (el: HTMLCalciteSheetElement) => Promise<void>;
+
+  /**
+   * Specifies the display mode - `"float"` (content is separated detached),
+   * or `"overlay"` (displays on top of center content).
+   */
+  @Prop({ reflect: true }) displayMode: DisplayMode = "overlay";
+
+  /** When `true`, disables the default close on escape behavior. */
+  @Prop({ reflect: true }) escapeDisabled = false;
+
+  /**
+   * When `position` is `"block-start"` or `"block-end"`, specifies the height of the component.
+   */
+  @Prop({ reflect: true }) heightScale: Scale = "m";
+
+  /**
+   * When `true`, prevents focus trapping.
+   */
+  @Prop({ reflect: true }) focusTrapDisabled = false;
+
+  @Watch("focusTrapDisabled")
+  handleFocusTrapDisabled(focusTrapDisabled: boolean): void {
+    if (!this.open) {
+      return;
+    }
+
+    focusTrapDisabled ? deactivateFocusTrap(this) : activateFocusTrap(this);
+  }
+
+  /**
+   * Specifies the label of the component.
+   */
+  @Prop() label!: string;
+
+  /** When `true`, displays and positions the component.  */
+  @Prop({ mutable: true, reflect: true }) open = false;
+
+  @Watch("open")
+  async toggleSheet(value: boolean): Promise<void> {
+    if (this.ignoreOpenChange) {
+      return;
+    }
+
+    onToggleOpenCloseComponent(this);
+    if (value) {
+      this.openSheet();
+    } else {
+      this.closeSheet();
+    }
+  }
+
+  /**
+   * We use an internal property to handle styles for when a modal is actually opened, not just when the open attribute is applied. This is a property because we need to apply styles to the host element and to keep the styles present while beforeClose is .
+   *
+   * @internal.
+   */
+  @Prop({ mutable: true, reflect: true }) opened = false;
+
+  /** When `true`, disables the closing of the component when clicked outside. */
+  @Prop({ reflect: true }) outsideCloseDisabled = false;
+
+  /** Determines where the component will be positioned. */
+  @Prop({ reflect: true }) position: LogicalFlowPosition = "inline-start";
+
+  /**
+   * This internal property, managed by a containing calcite-shell, is used
+   * to inform the component if special configuration or styles are needed
+   *
+   * @internal
+   */
+  @Prop() slottedInShell: boolean;
+
+  /**
+   * When `position` is `"inline-start"` or `"inline-end"`, specifies the width of the component.
```

Comment:
Doc question, does this affect height when the `writing-mode` is changed? We might need to fine-tune both `modal` and `sheet`'s doc separately depending on this.

Reply:
@macandcheese do you know the answer to this one?

Reply:
AFAIK it would still just set the container width to the supplied  default or overwritten css vars - maybe we could test for this? Any changes or doc could also apply to Shell Panel where we have similar width / height props / css vars.